src/text-transform/common.ts

```diff
@@ -0,0 +1,42 @@
+import {fullTransform, typografToHTML} from './utils';
+
+import {Lang} from '../utils/configure';
+
+export type ComplexItem = {[key: string]: string};
+export type Item = string | null | ComplexItem;
+export type Transformer = (text: string) => string;
+export type TransformerRaw = (lang: Lang, content: string) => string;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Parser<T = any> = (transformer: Transformer, block: T) => T;
+
+export const createItemsParser = (fields: string[]) => (transformer: Transformer, items: Item[]) =>
+    items.map((item) => {
+        if (!item) {
+            return item;
+        } else if (typeof item === 'string') {
+            return transformer(item);
+        } else {
+            return {
+                ...item,
+                ...fields.reduce<ComplexItem>((acc, fieldName) => {
+                    const result = {...acc};
```

Comment:
do we really need to copy acc here?

Reply:
Initially, I did not rewrite the logic, I left the previous version. I've corrected it now